### PR TITLE
add task progress page

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
# Pull Request Summary

## Changes
1. A new test file for `InputWithQuestions` has been added. The test file is named `inputWithQuestions.test.tsx` and is located in the `__tests__/Unit/Components/Progress` directory.
2. A new module for `ProgressForm` has been added. The module is named `ProgressForm.module.scss` and is located in the `src/components/ProgressForm` directory.
3. A new page for progress updates has been added. The page is named `[id].tsx` and is located in the `src/pages/progress` directory.
4. RTK will be added after confirming the API contracts and data transfer objects in a new PR.\
7. The feature is hidden behind a feature flag.
8. Screenshot of the Page
dev realdevsquad com_progress_OxYqJgf6Tyl90uci1mzs_dev=true (1)

Issues that I will be Working on after merging: https://github.com/Real-Dev-Squad/website-status/issues/533

To visit this page go to URL https://status.realdevsquad.com/progress/:taskId?dev=true

Test coverage report
image
image
image

### CODE REVIEW

File: __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { fireEvent, screen } from '@testing-library/react';
+
+describe('Progress form', function () {
+    it('Should Render 3 input fields with appropriate data', function (){
+        renderWithProviders(<ProgressForm/>);
+        const textBoxes = screen.getAllByRole('textbox');
+        expect(textBoxes).toHaveLength(3);
+        const button = screen.getByRole('button') as HTMLButtonElement;
+        expect(button).toBeInTheDocument();
+        expect(button.type).toBe('submit');
```
Comment (bhtibrewal) [inline __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx:12]: Suggestion: you can also write like this
`const button = screen.getByRole('button', {type: /submit/})`

File: __tests__/Unit/Components/ProgressForm/inputWithQuestions.test.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import InputWithQuestions from '@/components/ProgressForm/InputWithQuestions';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+describe.only('Input field with label', function () {
+    it('Should render a input field with appropriate label', function () {
+        render(<InputWithQuestions name="Test-Question" question="This is a test question"/>);
+        const input = screen.getByLabelText('Test-Question');
```
Comment (bhtibrewal) [inline __tests__/Unit/Components/ProgressForm/inputWithQuestions.test.tsx:7]:
```suggestion
const name = "Test-Question"
        render(<InputWithQuestions name={name} question="This is a test question"/>);
        const input = screen.getByLabelText(name);
```

File: src/components/ProgressForm/ProgressForm.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    {
+        id: 3,
+        name: 'blockers',
+        question: 'List down any blockers that you have',
+    },
+];
+
+const getFormattedDate = (): string => {
+    const date = new Date();
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressForm.tsx:25]: pass date as an argument and take this function to the utils folder. also write test for it not done yet. handle edge cases of date passed in different format, or not passed at all

File: src/components/ProgressForm/ProgressForm.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                <h2 className={styles.date}>on {getFormattedDate()}</h2>
+                {questions.map((question) => {
+                    return (
+                        <InputWithQuestions
+                            key={question.id}
+                            name={question.name}
+                            question={question.question}
+                        />
+                    );
+                })}
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressForm.tsx:72]:
```suggestion
                {questions.map((question) => 
                    (<InputWithQuestions
                            key={question.id}
                            name={question.name}
                            question={question.question}
                        />
                    )
               )}
```

File: src/components/ProgressForm/ProgressHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+function ProgressHeader() {
+    return (
+        <>
+            <header className={styles.header}>
+                <div>
+                <p>You have <span className={styles.mark}> 2 missed </span> Progress Updates</p>
+                <p>Lets try to avoid missing updates</p>
+                </div>
+            </header>
+        </>
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressHeader.tsx:12]: why do we need the react fragment here

File: src/components/ProgressForm/ProgressForm.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    return `${daysInWeek[date.getDay()]}, ${date.getDate()} ${
+        month[date.getMonth()]
+    } ${date.getFullYear()}`;
+};
+
+const ProgressForm = () => {
+    return (
+        <>
+            <NavBar />
+            <ProgressHeader/>
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressForm.tsx:60]: as the name of this component suggests it should only have the form, remove these
```
  <NavBar />
            <ProgressHeader/>
```

File: src/components/ProgressForm/ProgressHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+
+function ProgressHeader() {
+    return (
+        <>
+            <header className={styles.header}>
+                <div>
+                <p>You have <span className={styles.mark}> 2 missed </span> Progress Updates</p>
+                <p>Lets try to avoid missing updates</p>
+                </div>
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressHeader.tsx:10]: why is this extra uneccesary div added here?

File: src/components/taskDetails/index.tsx [hunk 8]
```diff
@@ -220,6 +219,13 @@ const TaskDetails: FC<Props> = ({ url, taskID }) => {
                         </section>
 
                         <section className={classNames.rightContainer}>
+                            <button
+                                onClick={() => {
+                                    document.location = `${STATUS_URL}/progress/${taskID}?dev=true`;
+                                }}
```
Comment (bhtibrewal) [inline src/components/taskDetails/index.tsx:225]: use router , not document.location

File: src/pages/tasks/[id].tsx [hunk 9]
```diff
@@ -6,11 +6,16 @@ const TaskDetailsPage = () => {
...
     if (!id) {
         return <PageNotFound />;
     }
 
-    return <TaskDetails url={TASK_DETAILS_URL} taskID={id} />;
+    return (
+        <>
+            <TaskDetails url={TASK_DETAILS_URL} taskID={id} />
+        </>
+    );
```
Comment (bhtibrewal) [inline src/pages/tasks/[id].tsx:19]: again `<></>` not needed

File: src/pages/progress/[id].tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import {useRouter} from 'next/router';
+import PageNotFound from '../404';
+import ProgressForm from '@/components/ProgressForm/ProgressForm';
+
+const ProgressUpdatesPage = () => {
+    const router = useRouter();
+    const id = router.query?.id as string;
+    const devMode = router.query?.dev as string;
```
Comment (bhtibrewal) [inline src/pages/progress/[id].tsx:8]: Destructure them in one line

**Review by bhtibrewal — CHANGES_REQUESTED** [review]

File: __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx (hunk 1 above)
Comment (RitikJaiswal75) [reply __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx:12]: As per the role API reference, there is no property named type

Official docs:
https://testing-library.com/docs/queries/byrole/#api

File: __tests__/Unit/Components/ProgressForm/inputWithQuestions.test.tsx (hunk 2 above)
Comment (RitikJaiswal75) [reply __tests__/Unit/Components/ProgressForm/inputWithQuestions.test.tsx:7]: done

File: src/components/ProgressForm/ProgressForm.tsx (hunk 4 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx:72]: done

File: src/components/ProgressForm/ProgressHeader.tsx (hunk 7 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressHeader.tsx:10]: Missed on that one

File: src/pages/progress/[id].tsx (hunk 10 above)
Comment (RitikJaiswal75) [reply src/pages/progress/[id].tsx:8]: done

File: src/components/ProgressForm/ProgressHeader.tsx (hunk 5 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressHeader.tsx:12]: removed the fragment

File: src/components/ProgressForm/ProgressForm.tsx (hunk 6 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx:60]: Do you want me to create a layout for it?

Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx:60]: Created a layout

File: src/components/taskDetails/index.tsx (hunk 8 above)
Comment (RitikJaiswal75) [reply src/components/taskDetails/index.tsx:225]: done

File: src/components/ProgressForm/ProgressForm.tsx (hunk 3 above)
Comment (bhtibrewal) [reply src/components/ProgressForm/ProgressForm.tsx]: Because if we need to format any other date apart from the latest date we can pass that date to it and reuse it.

Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx]: But it is just to get the latest date

**Review by shreya-mishra — DISMISSED** [review]
LGTM

File: src/utils/getLatestDate.ts [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+    const daysInWeek = [
+        'Sunday',
+        'Monday',
+        'Tuesday',
+        'Wednesday',
+        'Thursday',
+        'Friday',
+        'Saturday',
+    ];
```
Comment (bhtibrewal) [inline src/utils/getLatestDate.ts:26]: we shouldn't have this data in the function.

File: src/components/ProgressForm/ProgressForm.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        id: 2,
+        name: 'plan',
+        question: 'Planned progress before the next update',
+    },
+    {
+        id: 3,
+        name: 'blockers',
+        question: 'List down any blockers that you have',
+    },
+];
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressForm.tsx:21]: Move this out of this component
```suggestion
const questions = [
    {
        id: 1,
        name: 'progress',
        question: 'Task Progress after the previous update',
    },
    {
        id: 2,
        name: 'plan',
        question: 'Planned progress before the next update',
    },
    {
        id: 3,
        name: 'blockers',
        question: 'List down any blockers that you have',
    },
];
```

File: src/components/ProgressForm/ProgressForm.module.scss [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+$color-disabled: #e5e7eb;
+$diff-margin: $base-unit * 15;
+$similar-margin: $base-unit * 8;
+$section-padding-top-bottom: $base-unit * 30;
+$section-padding-left-right: $base-unit * 25;
+
+.header {
+    margin: $diff-margin auto;
+    background-color: $color-blue;
+    color: white;
```
Comment (bhtibrewal) [inline src/components/ProgressForm/ProgressForm.module.scss:14]: use color variable

File: src/components/ProgressForm/InputWithQuestions.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+
+interface inputProps {
+    name: string;
+    question: string;
+}
+
```
Comment (bhtibrewal) [inline src/components/ProgressForm/InputWithQuestions.tsx:8]: Make a common file for interface

File: src/utils/getLatestDate.ts [hunk 15]
```diff
@@ -0,0 +1,33 @@
+const getCurrentDate = (): string => {
```
Comment (bhtibrewal) [inline src/utils/getLatestDate.ts:1]: don't see test written for it

File: src/components/ProgressForm/ProgressForm.module.scss (hunk 13 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.module.scss:14]: I already have it as a variable

File: package.json [hunk 16]
```diff
@@ -9,7 +9,7 @@
     "dev": "run-p run-command:*",
     "build": "next build",
     "start": "next start -p $PORT",
-    "test": "jest --watch",
+    "test": "jest --coverage",
```
Comment (bhtibrewal) [inline package.json:12]: Please address this in a separate PR

File: src/utils/getLatestDate.ts (hunk 11 above)
Comment (RitikJaiswal75) [reply src/utils/getLatestDate.ts]: done

File: src/components/ProgressForm/ProgressForm.tsx (hunk 12 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx]: done

File: package.json (hunk 16 above)
Comment (RitikJaiswal75) [reply package.json]: done

File: src/utils/getLatestDate.ts (hunk 15 above)
Comment (RitikJaiswal75) [reply src/utils/getLatestDate.ts:3]: Yep it will go in next PR as already there are too many files in this one

File: src/components/ProgressForm/InputWithQuestions.tsx (hunk 14 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/InputWithQuestions.tsx:8]: Sure but it will increase the file count

Comment (RitikJaiswal75) [reply src/components/ProgressForm/InputWithQuestions.tsx]: done

File: __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx (hunk 1 above)
Comment (bhtibrewal) [reply __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx:12]: yes I mixed it up name , okay resolved

File: src/components/ProgressForm/ProgressForm.module.scss (hunk 13 above)
Comment (bhtibrewal) [reply src/components/ProgressForm/ProgressForm.module.scss:14]: Can you please point me to where you have defined it as a variable

**Review by bhtibrewal — COMMENTED** [review]
Please write the steps to see your feature in Preview, in the PR description

**Review by bhtibrewal — APPROVED** [review]
testing out the dismiss approval feature

File: src/components/ProgressForm/ProgressForm.module.scss [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+    border-radius: $base-unit * 2;
+}
+
+.buttonActive {
+    background-color: $color-blue;
+    color: white;
+    padding: $base-unit * 4 $base-unit * 8;
+    border-radius: $base-unit * 2;
+    margin-bottom: $diff-margin;
+    font-size: $base-unit * 4;
```
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.module.scss:66]: I am waiting on the design guidelines for that

[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.module.scss]: done

**Review by bhtibrewal — APPROVED** [review]

**Review by bhtibrewal — DISMISSED** [review]
Okay to merge

File: src/components/ProgressForm/ProgressForm.module.scss (hunk 13 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.module.scss]: sorry, my bad. Resolved them

File: src/components/ProgressForm/ProgressForm.tsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import getCurrentDate from '@/utils/getLatestDate';
+import InputWithQuestions from './InputWithQuestions';
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+import { questions } from '../constants/ProgressUpdates';
+
+const ProgressForm = () => {
+    return (
+        <form className={styles.form}>
+            <h1 className={styles.formHeading}>Task Updates</h1>
```
Comment (vvaibhavdesai) [inline src/components/ProgressForm/ProgressForm.tsx:9]: h1 in form ?

File: src/components/ProgressForm/ProgressForm.tsx [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+            <h2 className={styles.date}>on {getCurrentDate()}</h2>
```
Comment (vvaibhavdesai) [inline src/components/ProgressForm/ProgressForm.tsx:10]: h2 ?

File: src/components/ProgressForm/ProgressHeader.tsx [hunk 20]
```diff
@@ -0,0 +1,15 @@
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+
+function ProgressHeader() {
+    return (
+        <header className={styles.header}>
+            <p>
+                You have <span className={styles.mark}> 2 missed </span>{' '}
```
Comment (vvaibhavdesai) [inline src/components/ProgressForm/ProgressHeader.tsx:7]: `{' '}` ???

Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressHeader.tsx]: Missed on this, thanks for pointing out

File: src/components/ProgressForm/ProgressForm.tsx (hunk 18 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx]: fixedthis

File: src/components/ProgressForm/ProgressForm.tsx (hunk 19 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx]: fixed this

**Comment by iamYashSinha** [comment]
Please add the Figma design link as well if any.

File: src/components/ProgressForm/InputWithQuestions.tsx [hunk 21]
```diff
@@ -0,0 +1,22 @@
+import { useState } from 'react';
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+import { inputPropsTypes } from '@/types/ProgressUpdates';
+
+const InputWithQuestions = ({ name, question }: inputPropsTypes) => {
+    const [answer, setAnswer] = useState('');
+    return (
+        <div className={styles.inputComponent}>
+            <label className={styles.label} htmlFor={name} aria-label={name}>
```
Comment (vvaibhavdesai) [inline src/components/ProgressForm/InputWithQuestions.tsx:9]: what is the name here in aria-label ?

File: src/components/ProgressForm/ProgressForm.tsx [hunk 22]
```diff
@@ -0,0 +1,22 @@
+import InputWithQuestions from './InputWithQuestions';
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+import { questions } from '../constants/ProgressUpdates';
+
+const ProgressForm = () => {
+    return (
+        <form className={styles.form}>
+            {questions.map((question) => (
```
Comment (vvaibhavdesai) [inline src/components/ProgressForm/ProgressForm.tsx:8]: take this out as a constant

File: src/components/taskDetails/index.tsx [hunk 23]
```diff
@@ -218,6 +220,13 @@ const TaskDetails: FC<Props> = ({ url, taskID }) => {
                         </section>
 
                         <section className={classNames.rightContainer}>
+                            <button
+                                onClick={() =>
+                                    router.push(`/progress/${taskID}?dev=true`)
```
Comment (vvaibhavdesai) [inline src/components/taskDetails/index.tsx:225]: take this out as a function

File: src/pages/progress/[id].tsx [hunk 24]
```diff
@@ -0,0 +1,17 @@
+import { useRouter } from 'next/router';
+import PageNotFound from '../404';
+import ProgressLayout from '@/components/ProgressForm/ProgressLayout';
+
+const ProgressUpdatesPage = () => {
+    const router = useRouter();
+
+    const { id, dev } = router.query;
+
+    if (!id || dev !== 'true') {
```
Comment (vvaibhavdesai) [inline src/pages/progress/[id].tsx:10]: can you explain what is this condition for ?

**Review by vvaibhavdesai — CHANGES_REQUESTED** [review]
image
have added a bunch of comments also the above banner looks a lot big can we make it small ?

File: src/components/ProgressForm/ProgressForm.tsx (hunk 22 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx:8]: what constant? didn't get you, questions is already a constant

File: src/components/ProgressForm/InputWithQuestions.tsx (hunk 21 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/InputWithQuestions.tsx:9]: it would be the question name

File: src/components/ProgressForm/ProgressLayout.tsx [hunk 25]
```diff
@@ -0,0 +1,22 @@
...
+import ProgressForm from './ProgressForm';
+import styles from '@/components/ProgressForm/ProgressForm.module.scss';
+import getCurrentDate from '@/utils/getLatestDate';
+
+function ProgressLayout() {
+    return (
+        <>
+            <NavBar />
+            <ProgressHeader />
+            <div className={styles.container}>
```
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressLayout.tsx:13]: Section for? I don't think the form should go in as a section

File: src/pages/progress/[id].tsx (hunk 24 above)
Comment (RitikJaiswal75) [reply src/pages/progress/[id].tsx:10]: the feature is behind the feature flag dev=true I am checking if that is passed at the same time I am also checking if the id is present in the URL, if any of this is absent we show a 404 page

File: src/components/taskDetails/index.tsx (hunk 23 above)
Comment (RitikJaiswal75) [reply src/components/taskDetails/index.tsx:225]: Function just for 1 line?

**Comment by RitikJaiswal75** [comment]
> have added a bunch of comments also the above banner looks a lot big can we make it small?
@vvaibhavdesai the header is as per the design guidelines also the feature is behind a feature flag, Minor UI changes can get in later.

File: src/components/ProgressForm/ProgressHeader.tsx [hunk 26]
```diff
@@ -0,0 +1,15 @@
...
+    return (
+        <header className={styles.header}>
+            <p>
+                You have <span className={styles.mark}> 2 missed</span> Progress
+                Updates
+            </p>
+            <p>Lets try to avoid missing updates</p>
+        </header>
+    );
+}
```
Comment (heyrandhir) [inline src/components/ProgressForm/ProgressHeader.tsx:13]: we should conditionally render this component based on whether there is a update missed or not. `<p>Lets try to avoid missing updates</p>` can be removed if no progress updates are missed.

**Review by heyrandhir — CHANGES_REQUESTED** [review]
Overall looks great. good work. requested some minor changes.

File: src/components/ProgressForm/ProgressForm.tsx (hunk 22 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx:8]: image
Is this what you are trying to say.

File: __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx [hunk 27]
```diff
@@ -0,0 +1,33 @@
+import ProgressForm from '@/components/ProgressForm/ProgressForm';
+import { renderWithProviders } from '@/test-utils/renderWithProvider';
+import { fireEvent, screen } from '@testing-library/react';
```
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx:3]: Yes, that is going to come in future PRS where I will be fixing up the import structure. Currently, we are not following any structure in our imports. I will soon fix them. in consecutive PRs

File: src/components/ProgressForm/ProgressHeader.tsx (hunk 26 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressHeader.tsx:13]: Conditional rendering and form validations will come once the API contracts are fixed

File: __tests__/Unit/Components/ProgressForm/inputWithQuestions.test.tsx [hunk 28]
```diff
@@ -0,0 +1,30 @@
+import InputWithQuestions from '@/components/ProgressForm/InputWithQuestions';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+const name = 'Test-Question';
+
+describe.only('Input field with label', function () {
```
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply __tests__/Unit/Components/ProgressForm/inputWithQuestions.test.tsx:6]: Thanks for pointing it out, I fixed it.

File: __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx (hunk 27 above)
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply __tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx]: done, since it would save the effort when the whole site will be going through the refactor.

File: src/components/ProgressForm/ProgressForm.tsx (hunk 22 above)
Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressForm.tsx]: As per the latest commit I have used useReducer and with the component having a dependency on the reducer, I can not bring it out of the component.

File: src/components/taskDetails/index.tsx (hunk 23 above)
Comment (RitikJaiswal75) [reply src/components/taskDetails/index.tsx:225]: Since it uses Router I need to use router.push inside a component.

**Review by heyrandhir — DISMISSED** [review]

File: src/components/ProgressForm/ProgressLayout.tsx [hunk 29]
```diff
@@ -0,0 +1,26 @@
...
+
+import { questions } from '@/constants/ProgressUpdates';
+
+function ProgressLayout() {
+    return (
+        <>
+            <NavBar />
+            <ProgressHeader />
+            <section className={styles.container}>
+                <h1 className={styles.formHeading}>Task Updates</h1>
```
Comment (vvaibhavdesai) [inline src/components/ProgressForm/ProgressLayout.tsx:18]: h1 shouldnt be inside a tag

Comment (RitikJaiswal75) [reply src/components/ProgressForm/ProgressLayout.tsx]: done

**Review by vvaibhavdesai — APPROVED** [review]

**Review by heyrandhir — APPROVED** [review]

**Review by bhtibrewal — APPROVED** [review]
LGTM
